### PR TITLE
update: TopBar refactor

### PR DESCRIPTION
**Pull request by kgajowy**
Refactored TopBar::
* added `stage-0` to support fat arrows
* removed unnecessary proprs (onClicks)
* added the code so that the actions login/logout are dispatched properly (which should be rather renamed later to only do navigation/modal probaly)
* rewrote the tests to be per-context
* added more tests (clicking buttons, dispatching actions)
* moved all tests to `__tests__` so that they are not mixed with real components.

### CODE REVIEW

File: src/components/TopBar.js [hunk 1]
```diff
@@ -7,28 +7,29 @@ import Text from './Text';
...
-        const buttonAction = loggedUser ? logout : login;
+        const {loggedUser} = this.props;
         const buttonText = loggedUser ? 'Wyloguj' : 'Zaloguj';
         return (
             <TopBarWrapper>
                 <Text type="basic" color={ Colors.white } display={ 'inline' }>
                     Zaloguj się przez GitHub
                 </Text>
-                <Button marginLeft='10px' onClick={ buttonAction }
+                <Button marginLeft='10px' onClick={this.onClick}
```
Comment (ulaulala) [inline src/components/TopBar.js]: This line is sad without spaces :(

Comment (piecioshka) [reply src/components/TopBar.js]: why @kgajowy why 😢

Comment (kgajowy) [reply src/components/TopBar.js]: We should force this style to let developers work like they want ;) I will make a prettier+husky integration in the next PR!

Comment (piecioshka) [reply src/components/TopBar.js]: maybe try do it in that branch, hmm?

Comment (kgajowy) [reply src/components/TopBar.js]: no, rather not - feature branches <3

**Pull request merged**